streamwebs_frontend/streamwebs/tests/models/test_canopy_cover.py

```diff
@@ -0,0 +1,144 @@
+
+    def test_datasheet_CreateCanopyCover(self):
+        """Tests that a Canopy Cover object is actually created, checks that
+           the correct (shaded) value is received from the CC_Cardinal model"""
+        default_dt = timezone.now()
+
+        site = Site.objects.create_site('test', 'some_type', 'some_slug')
+
+        north = CC_Cardinal.objects.create_shade(True, True, False, False,
+                                                 False, True, False, False,
+                                                 True, True, True, False, True,
+                                                 False, False, False, True,
+                                                 False, True, True, False,
+                                                 True, False, False, 11)
+
+        east = CC_Cardinal.objects.create_shade(True, True, False, False,
+                                                False, True, False, False,
+                                                True, True, True, False, True,
+                                                False, False, False, False,
+                                                False, True, True, False,
+                                                False, False, False, 8)
+
+        south = CC_Cardinal.objects.create_shade(True, True, True, False,
+                                                 False, True, False, False,
+                                                 True, True, True, True, True,
+                                                 False, False, False, True,
+                                                 False, True, True, False,
+                                                 True, False, True, 14)
+
+        west = CC_Cardinal.objects.create_shade(True, True, False, True,
+                                                False, True, True, False,
+                                                True, True, True, True, True,
+                                                False, False, True, True,
+                                                False, True, True, True, True,
+                                                True, False, 17)
+
+        canopyc = Canopy_Cover.objects.create(school='School A',
+                                              date_time=default_dt, site=site,
+                                              weather='cloudy', north=north,
+                                              east=east, south=south,
+                                              west=west, est_canopy_cover=50)
+
+        self.assertEqual(canopyc.school, 'School A')
+        self.assertEqual(canopyc.date_time, default_dt)
+        self.assertEqual(canopyc.weather, 'cloudy')
+        self.assertEqual(canopyc.north.shaded, 11)
+        self.assertEqual(canopyc.east.shaded, 8)
+        self.assertEqual(canopyc.south.shaded, 14)
+        self.assertEqual(canopyc.west.shaded, 17)
+        self.assertEqual(canopyc.est_canopy_cover, 50)
```

Comment:
If you want, you can create a list of all the True/False values like so:
```python
bools = [True, True, False, True...]
```
and use that to create the shades like so:

```python
north = CC_Cardinal.objects.create_shade(*bools, 11)
```

That way, you can [loop through each field](http://stackoverflow.com/questions/10353804/how-do-i-loop-through-fields-of-an-object) in `canopyc.north` and compare against that position in `bools`.